src/modules/flow/http-server/http-server.c

```diff
@@ -295,44 +295,45 @@ string_response_cb(void *data, struct sol_http_request *request)
     } else {
         r = asprintf(&response_str, "%s", mdata->value.s);
     }
-    SOL_INT_CHECK_GOTO(r, < 0, end);
-
-    r = sol_http_param_add(&response.param, SOL_HTTP_REQUEST_PARAM_HEADER(
-        HTTP_HEADER_CONTENT_TYPE, (send_json) ? HTTP_HEADER_CONTENT_TYPE_JSON : HTTP_HEADER_CONTENT_TYPE_TEXT));
-    SOL_INT_CHECK_GOTO(r, != true, end_response);
-
-    r = sol_buffer_set_slice(&response.content, sol_str_slice_from_str(response_str));
-    SOL_INT_CHECK_GOTO(r, < 0, end_response);
+    SOL_INT_CHECK(r, < 0, r);
 
-    r = sol_http_server_send_response(request, &response);
-    SOL_INT_CHECK_GOTO(r, < 0, end);
+    r = sol_buffer_set_slice(content, sol_str_slice_from_str(response_str));
+    free(response_str);
+    return r;
+}
 
-    if (method == SOL_HTTP_METHOD_POST) {
-        sol_flow_send_string_packet(mdata->node, SOL_FLOW_NODE_TYPE_HTTP_SERVER_STRING__OUT__OUT, mdata->value.s);
+static int
+string_post_cb(struct http_data *mdata, struct sol_flow_node *node,
+    struct sol_http_param_value *value)
+{
+    if (streq(value->value.key_value.key, "value")) {
+        free(mdata->value.s);
+        mdata->value.s = strdup(value->value.key_value.value);
+        if (!mdata->value.s) {
+            SOL_WRN("Could not duplicate the string");
+            mdata->value.s = NULL;
+            return -ENOMEM;
+        }
+    } else {
+        return -EINVAL;
     }
```

Comment:
btw, this code could go in sol `sol-util.h` so more places could do this, like some string nodes (@glima).